engine/lib/elgglib.php

```diff
@@ -1211,6 +1211,39 @@ function elgg_extract_class(array $array, $existing = [], $extract_key = 'class'
 	return array_values(array_unique($existing));
 }
 
+/**
+ * Calls a callable passing arguments and applying logic based on flags
+ *
+ * @param callable $callable  Callable to call
+ * @param array    $arguments Arguments to pass to callable
+ * @param null     $flags     Flags
+ *
+ * @return mixed
+ */
+function elgg_call(callable $callable, $arguments = null, $flags = null) {
+
+	$ia = elgg_get_ignore_access();
+	if ($flags & ELGG_IGNORE_ACCESS) {
+		elgg_set_ignore_access(true);
+	} else if ($flags & ELGG_ENFORCE_ACCESS) {
+		elgg_set_ignore_access(false);
+	}
+
+	$ha = access_get_show_hidden_status();
+	if ($flags & ELGG_SHOW_DISABLED_ENTITIES) {
+		access_show_hidden_entities(true);
+	} else if ($flags & ELGG_HIDE_DISABLED_ENTITIES) {
+		access_show_hidden_entities(false);
+	}
+
+	$result = call_user_func_array($callable, (array) $arguments);
```

Comment:
For this to be bulletproof you need to catch exceptions, reset access, and I guess re-throw them.